### PR TITLE
docs: add example to provider.ts in utils

### LINKED ISSUE #1090: Improve JSDocs by adding examples of usages
Inspired by Viem docs, example:

https://viem.sh/docs/accounts/privateKey#usage

This is a big endeavor so it can be completed in multiple smaller PR-s (and more users can take it in parallel!)

IMPORTANT: add `@example` only to **exported** functions 

TODO utils:
- [ ] src/utils/calldata/validate 
- [ ] src/utils/calldata/formatter
- [ ] src/utils/calldata/propertyOrder
- [ ] src/utils/calldata/requestParser
- [ ] src/utils/calldata/responseParser
- [ ] src/utils/calldata/tuple
- [ ] src/utils/calldata/parser/index.ts
- [ ] src/utils/events/index.ts
- [ ] src/utils/hash/classHash.ts
- [ ] src/utils/hash/transactionHash/v3.ts

### PR DESCRIPTION
**Pull request by vibenedict**
## Motivation and Resolution
Closes #1090 
...

### RPC version (if applicable)

...

## Usage related changes

- Change 1.
- ...

## Development related changes

- Change 1.
- ...

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

**Comment by vibenedict** [comment]
Done

**Comment by ivpavici** [comment]
@vibenedict thanks for resolving conflicts, but my previous comment is not yet addressed, regardingg adding missing `@param` on all functions?

**Review by PhilippeR26 — CHANGES_REQUESTED** [review]
As explained in Telegram https://t.me/c/2066195975/59 ,
please follow very rigorously  the template of this PR #1100, especially for the examples : result always in a `result` variable, result displayed in a comment (`//`);

**Comment by vibenedict** [comment]
@ivpavici @PhilippeR26

File: src/utils/provider.ts [hunk 1]
```diff
@@ -51,6 +76,25 @@ export function createSierraContractClass(contract: CompiledSierra): SierraContr
  * Create Contract Class from a given CompiledContract or string
  *
  * (CompiledContract or string) -> ContractClass
+ * 
+ * @param {CompiledContract | string} contract - Compiled contract or string
+ *
```
Comment (ivpavici) [inline src/utils/provider.ts:81]: missing returns

File: src/utils/provider.ts [hunk 2]
```diff
@@ -37,6 +43,25 @@ export function wait(delay: number) {
  * Create Sierra Contract Class from a given Compiled Sierra
  *
  * CompiledSierra -> SierraContractClass
+ * 
+ * @param {CompiledSierra} contract - Compiled sierra code
+ *
```
Comment (ivpavici) [inline src/utils/provider.ts:48]: missing returns

File: src/utils/provider.ts [hunk 3]
```diff
@@ -26,6 +26,12 @@ import type { GetTransactionReceiptResponse } from './transactionReceipt';
 
 /**
  * Helper - Async Sleep for 'delay' time
+ *
+ * @param {number} delay - Number of milliseconds to delay.
+ * @example
```
Comment (ivpavici) [inline src/utils/provider.ts:31]: missing returns

**Review by ivpavici — CHANGES_REQUESTED** [review]

File: src/utils/provider.ts [hunk 4]
```diff
@@ -26,6 +26,12 @@ import type { GetTransactionReceiptResponse } from './transactionReceipt';
...
+ * ```typescript
+ * wait(1000) // 1000 milliseconds == 1 second
```
Comment (PhilippeR26) [inline src/utils/provider.ts:33]: It's an async function.
You have to add the `await` prefix.
All the functions of this file are in the `provider` space. To correct.

File: src/utils/provider.ts [hunk 5]
```diff
@@ -51,6 +76,25 @@ export function createSierraContractClass(contract: CompiledSierra): SierraContr
...
+        "0x0",
+        "0x2",
+        "0x4",
+        "0x1",
+        "0x3b4",
+        "0x4c",
+        "0x65",
+        "0x52616e6765436865636b",...})
+ * // result = {sierra_program: 'H4sIAAAAAAAAA6x9WZbsrI7uVGqd53qgb8ZynwzYY7jDv5JAAmxHZuQ+96yq/L0jIzEINZ8axP/5j/q/+j//+z/wH9f/o/p/zPbh+Iot49+u9v8G3//rTdDhDDF4Z0MKPthQ+m+S2v6n1S//638VvdXW2PQ6RvxuDG+jiybCXKJ7Hef6ZRi9E+Q89WmKLilfqbrsL6PUCf8...}
+ * ```
```
Comment (PhilippeR26) [inline src/utils/provider.ts:97]: Last function hasn't been made.

File: src/utils/provider.ts [hunk 6]
```diff
@@ -37,6 +43,25 @@ export function wait(delay: number) {
...
```
Comment (PhilippeR26) [inline src/utils/provider.ts:64]: `createSierraContractClass` and `parseContract` are performing the opposite action of the other, but you have the same example and result for both cases.
You should re-test these examples ; you will quickly find your error.

**Review by PhilippeR26 — CHANGES_REQUESTED** [review]
Some work remains.

**Pull request closed without merging**